cpp/tests/utilities/column_wrapper.cuh

```diff
@@ -0,0 +1,396 @@
+      host_data[row] = value_initalizer(row);
+
+      if (true == bit_initializer(row)) {
+        gdf::util::turn_bit_on(host_bitmask.data(), row);
+      }
+    }
+    initialize_with_host_data(host_data, host_bitmask);
+  }
+
+  /**---------------------------------------------------------------------------*
+   * @brief Returns a pointer to the underlying gdf_column.
+   *
+   *---------------------------------------------------------------------------**/
+  gdf_column* get() { return &the_column; }
+  gdf_column const* get() const { return &the_column; }
+
+  /**---------------------------------------------------------------------------*
+   * @brief Returns the null count of the underlying column.
+   *
+   *---------------------------------------------------------------------------**/
+  gdf_size_type null_count() const { return the_column.null_count; }
+
+  /**---------------------------------------------------------------------------*
+   * @brief Copies the underying gdf_column's data and bitmask to the host.
+   *
+   * Returns a tuple of two std::vectors. The first is the column's data, and
+   * the second is the column's bitmask.
+   *
+   *---------------------------------------------------------------------------**/
+  auto to_host() const {
+    gdf_size_type const num_masks{gdf_get_num_chars_bitmask(the_column.size)};
+    std::vector<ColumnType> host_data;
+    std::vector<gdf_valid_type> host_bitmask;
+
+    if (nullptr != the_column.data) {
+      // TODO Is there a nicer way to get a `std::vector` from a device_vector?
+      host_data.resize(the_column.size);
+      CUDA_RT_CALL(cudaMemcpy(host_data.data(), the_column.data,
+                              the_column.size * sizeof(ColumnType),
+                              cudaMemcpyDeviceToHost));
+    }
+
+    if (nullptr != the_column.valid) {
+      host_bitmask.resize(num_masks);
+      CUDA_RT_CALL(cudaMemcpy(host_bitmask.data(), the_column.valid,
+                              num_masks * sizeof(gdf_valid_type),
+                              cudaMemcpyDeviceToHost));
+    }
+
+    return std::make_tuple(host_data, host_bitmask);
+  }
+
+  /**---------------------------------------------------------------------------*
+   * @brief Prints the values of the underlying gdf_column.
+   *
+   *---------------------------------------------------------------------------**/
+  void print() const {
+    // TODO Move the implementation of `print_gdf_column` here once it's removed
+    // from usage elsewhere
+    print_gdf_column(&the_column);
+  }
+
+  /**---------------------------------------------------------------------------*
+   * @brief Functor for comparing if two elements between two gdf_columns are
+   * equal.
+   *
+   *---------------------------------------------------------------------------**/
+  struct elements_equal {
+    gdf_column lhs_col;
+    gdf_column rhs_col;
+    bool nulls_are_equivalent;
```

Comment:
We need to check whether the SQL standard we conform to has NULLs equal. In ISO SQL I'm pretty sure they are.

Reply:
I'm not really concerned either way, which is why I left it as a parameter.

Furthermore, this is just a unit testing utility, not something that is used in production features and is therefore not end-user consumable.